### PR TITLE
Feat/btv breakpoints MOVEMENT-2032

### PR DESCRIPTION
**Pull request by asdo6rs**

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import {NodeManager} from './utils';
+
+// NodeManager.getInstance();
```
Comment (tonygaetani) [inline lib/index.ts:3]: remove this

File: lib/nodes/Base.ts [hunk 2]
```diff
@@ -152,6 +163,17 @@ export class Base<S extends BlueshellState, E> implements BaseNode<S, E> {
 		this._parent = path;
 	}
 
+	public get parent() {
+		return this._parent;
+	}
+
+	public get id(): string {
+		if (!this._id) {
+			this._id = `n${v4().replace(/\-/g, '')}`;
```
Comment (tonygaetani) [inline lib/nodes/Base.ts:172]: what'st he `n` for?

File: lib/models/BaseNode.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import {BlueshellState, NodeStorage, ResultCode} from '../models';
 export interface BaseNode<S extends BlueshellState, E> {
 	name: string;
 	parent: string;
+	id: string;
```
Comment (tonygaetani) [inline lib/models/BaseNode.ts:9]: is this readonly? or should there be a setter in Base.ts

File: lib/utils/archyTree.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import * as archy from 'archy';
+import archy = require('archy');
```
Comment (tonygaetani) [inline lib/utils/archyTree.ts:1]: why this change?

File: lib/utils/nodeManager.ts [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+						reject(err);
+					} else {
+						resolve();
+					}
+				});
+			} else {
+				resolve();
+			}
+		});
+	}
```
Comment (tonygaetani) [inline lib/utils/nodeManager.ts:421]:
```suggestion
	public async shutdown() {
		if (this.server) {
			await util.promisify(this.server.close.bind(this.server));
		}
	}
```

I think this would work and be more concise, but I see no problem with the way it's implemented here anyway

**Review by tonygaetani — COMMENTED** [review]
I have some questions but it's minor stuff, overall LGTM

File: lib/index.ts (hunk 1 above)
Comment (asdo6rs) [reply lib/index.ts:3]: done (next push)

File: lib/nodes/Base.ts (hunk 2 above)
Comment (asdo6rs) [reply lib/nodes/Base.ts:172]: @mbittarelli6river it looks like this was in there from the beginning of when we added it in dec 2020 - is that to make it start with a string instead of a number?

File: lib/models/BaseNode.ts (hunk 3 above)
Comment (asdo6rs) [reply lib/models/BaseNode.ts:9]: Should be read-only.  Updated (next push)

File: lib/utils/archyTree.ts (hunk 4 above)
Comment (asdo6rs) [reply lib/utils/archyTree.ts:1]: While playing around with some ts settings, this change had to be made, but it looks like our current config allows us to keep the original import without require.  Reverting (next push).

File: lib/utils/nodeManager.ts (hunk 5 above)
Comment (asdo6rs) [reply lib/utils/nodeManager.ts:421]: That is much more concise - since we're making some fixes anyways, I included this change as well (next push)

File: lib/nodes/Base.ts (hunk 2 above)
Comment (mbittarelli6river) [reply lib/nodes/Base.ts:172]: it was holdover from https://github.com/6RiverSystems/blueshell/pull/147/files#diff-f48473d5b5c7f55d617959eaa22e2bdeac04ddd3fd4d17f7d2bc711a94e6e4b9L60 :)

Comment (tonygaetani) [reply lib/nodes/Base.ts:172]: oh it's always been like that? nevermind then

**Review by tonygaetani — APPROVED** [review]
